### PR TITLE
Return ZLP on empty requests of descriptors 

### PR DESCRIPTION
**Pull request by twam**

### CODE REVIEW

**Comment by straithe** [comment]
Thank you for your Pull Request! I have assigned a reviewer to this issue and it may take them some time to get to reviewing this. In the meantime, would you be willing to work on addressing the conflicts with this PR?

**Comment by twam** [comment]
@straithe sure: Rebased on lastest main.

**Comment by straithe** [comment]
Thank you very much! I do have a reviewer assigned to this PR. It might take them a bit of time to get to this, but please know that we have seen this PR and appreciate your work.

**Review by miek — CHANGES_REQUESTED** [review]
Thanks for the PR!

I've added a couple of comments inline & made the proposed changes here: https://github.com/miek/luna/commit/822423d26e34b1e4c27ac73374a90b8e57eddf65. Let me know what you think.

**Comment by twam** [comment]
Checked your suggestion solution and really like it! Suggest to merge that instead.

File: luna/gateware/usb/usb2/descriptor.py [hunk 1]
```diff
@@ -438,7 +439,7 @@ def elaborate(self, platform) -> Module:
                 # ... apply our start position...
                 m.d.sync += position_in_stream.eq(self.start_position),
 
-                is_valid_send = (length > 0)
+                is_valid_send = (length > 0) | (self.start_position > 0)
```
[the remark this answers is not in the document]
Comment (twam) [reply luna/gateware/usb/usb2/descriptor.py:442]: Good point. Haven't checked that, just extended. :)

**Comment by miek** [comment]
Merged manually with those changes, thanks again!

**Pull request closed without merging**